hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/BalancerConditionals.java

```diff
@@ -90,8 +94,20 @@ boolean isReplicaDistributionEnabled() {
       .anyMatch(DistributeReplicasConditional.class::isAssignableFrom);
   }
 
-  boolean shouldSkipSloppyServerEvaluation() {
-    return isConditionalBalancingEnabled();
+  boolean isTableIsolationEnabled() {
+    return conditionalClasses.contains(MetaTableIsolationConditional.class);
```

Comment:
This strategy of feature enablement by class presence is interesting. I wonder how we'll have to change this approach if we introduce user-provided implementations in the future. Just a thought.

Reply:
When I introduce SystemTableIsolation I anticipate making this an isAssignableFrom check much like `isReplicaDistributionEnabled`